src/Compilers/VisualBasic/Portable/Syntax/VisualBasicSyntaxNode.vb

```diff
@@ -474,6 +474,24 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return SyntaxTree.GetDiagnostics(Me)
         End Function
 
+        Private Protected Overrides Function IsSimilarToCore(other As SyntaxNode) As Boolean
```

Comment:
i'm a bit wary of these going directly on SyntaxNode.  i feel like it would be better to just be in the syntaxdiffer.  that may be a bit annoying as we'd need to make SyntaxDiffer inheritable, but i do think like it would be better to isolate tehre rather than have this particular one-off here. 

Particularly: the primary issue i have is that SyntaxNode isn't wishy-washy :)  It's a strictly defined, highly invariant-based API.  I'd like it to stay that way.  Well defined, with clear semantics on how every part of it should behave.  However, SyntaxDiffer, but it's very nature is going to make a 'best effort' on doing a diff since htere's no perfect algorithm out there.  So having this sort of logic go in there just seems more sensible to me.

Reply:
@CyrusNajmabadi I would like to come back to this PR, but I'm having trouble figuring out a good way to separate this code from the language-specific `SyntaxNode`, while keeping it in the language-specific assembly (because it needs access to `SyntaxKind`).

The best approach I could think of is if `SyntaxNode` gave access to the correct `SyntaxDiffer`. In other words, `CSharpSyntaxNode` would contain:

```c#
internal override SyntaxDiffer GetDiffer(SyntaxNode newNode, bool computeNewText) =>
    new CSharpSyntaxDiffer(this, newNode, computeNewText);
```

Do you think this would be a good solution? If not, what do you suggest?

Reply:
@svick That seems reasonable to me.  You can have an abstract base class with core logic, and any lang specific logic can go in the derived types.